python/ray/tune/tune.py

```diff
@@ -706,6 +741,16 @@ class and registered trainables.
 
     progress_metrics = _detect_progress_metrics(_get_trainable(run_or_experiment))
 
+    # setting the right context for console logging
+    is_single_trial = search_alg.total_samples == 1
+
+    def is_rllib():
+        from ray.rllib.algorithms.algorithm import Algorithm
+
+        return issubclass(_get_trainable(run_or_experiment), Algorithm)
+
+    set_execution_type(is_rllib(), is_single_trial)
```

Comment:
i think it's odd that we have this `_execution_type`, which sorta breaks abstraction boundary between train/tune/rllib. 

Can we somehow push this to the downstream libraries as a generic interface? i.e., trainable.logger_format or something

Reply:
yeah, that's fair. Think of this PR more about the structure of ProgressReporter stuff and the priorities of different verbosity settings. 
where to get this information - yeah we can expose an API to ask Trainable to specify that. Although I think it may be easier right now to do it this way (gather in tune.run) given that a lot of tune stuff is very catered towards to rllib stuff and it's a lot more if we want to push the abstraction to the right level.

Reply:
i.e. we need to collect some more context and a survey of current coupling between tune and rllib to decide what this interface should look like. Which I think is orthogonal to the current console output thing.